Update life.js to fix DNDBeyond Updates

SupremeDevice opened the pull request:
Fixes #1 
(Not the multi-classing bit but it does get it at least working a bit)

SupremeDevice committed:
Update life.js to fix DNDBeyond Updates

donoftime commented on life.js:
```diff
@@ -1,9 +1,10 @@
 const lifeBeyond = async () => {
   const character = await fetchCharacter();
-  const lifeDomain = character.classes.filter((characterClass) => characterClass.subclassDefinition.name == 'Life Domain')[0];
+  const lifeDomain = character.classes.filter((charClass) => charClass.subclassDefinition.name == 'Life Domain')[0];
```
In this case, `characterClass` is a variable name specified by us, rather than something that was specified by dndbeyond in their DOM. So this does not need to change.

For more information on this syntax, please see the [ES6 arrow function page on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) which I think gives a nice explanation.

donoftime commented on life.js:
```diff
@@ -1,9 +1,10 @@
   if (lifeDomain == null) {
     return;  // nothing to do here
   }
+  await new Promise(resolve => setTimeout(resolve, 3000));
```
Right here is where the core of the problem occurs. 

I looked into what is happening as the dndbeyond application initializes, and from what I see they are no longer loading the page all at once. Now, they are loading individual components one at a time -- I assume to improve the perceived page-load times. 

Before the update, our request for character data would have resolved long after the sidebar was on the screen. But now, after we get the character data from the backend, we try to begin observing the sidebar in the DOM only to find there is no sidebar on the screen!

The setTimeout here addresses that timing problem by providing a little 3 second pause before moving on to attach the spell book observer.

🤔 However, I generally try to avoid using time-based delays (aka "sleeps") as they are not always dependable. While the sidebar may be added to one person's screen after 2 seconds, for another person it may be 4 seconds and they would see the same problem again.

So instead of a time-based delay, I will put together a commit to add an event-based delay which watches the `document.body` for the sidebar to appear before attaching the spell book observer.

Commit:
fix(load): Wait for sidebar before attaching spellbook observer

The dndbeyond app is now initializing in a way that we would finish loading the character data
before the sidebar was actually placed on the screen. This would cause the observer to fail and no
longer update the spellbook as people opened healing spells. This change adds another observer, this
time to the document body, which watches for the sidebar to appear before creating the spellbook
observer.

fix #1

donoftime commented:
I created a new character who is a Life Cleric with the Monk subclass to test, and with the changes above it seems to be working now 🙂 

I expect the issue you saw regarding subclassing was also related to the delays which occur while dndbeyond does its initial page load. 

I will go ahead and merge your PR and get it deployed to the chrome store
<img width="883" alt="image" src="https://user-images.githubusercontent.com/5498070/68072039-8eb94780-fd57-11e9-92d5-b37cf96c4df9.png">
